File: data/readable_format/fortran/file772.f90
PROGRAM testST_LCUC
    IMPLICIT NONE
    CHARACTER(len=20) :: string, outstr
    INTEGER :: iret

    ! Test data
    string = 'Hello, World!'
    outstr = ' ' ! Initialize with spaces
    iret = -1    ! Initialize with a non-zero value

    ! Call the subroutine
    CALL ST_LCUC(string, outstr, iret)

    ! Check the results
    IF (TRIM(outstr) == 'HELLO, WORLD!' .AND. iret == 0) THEN
        PRINT *, 'Test passed.'
    ELSE
        PRINT *, 'Test failed.'
    END IF

END PROGRAM testST_LCUC

SUBROUTINE ST_LCUC(string, outstr, iret)
    IMPLICIT NONE
    CHARACTER(len=*), INTENT(IN) :: string
    CHARACTER(len=*), INTENT(OUT) :: outstr
    INTEGER, INTENT(OUT) :: iret
    INTEGER :: i, j, isize

    iret = 0
    outstr = string
    isize = LEN_TRIM(outstr)

    DO i = 1, isize
        IF ((outstr(i:i) >= 'a') .AND. (outstr(i:i) <= 'z')) THEN
            j = ICHAR(outstr(i:i))
            j = j - 32
            outstr(i:i) = CHAR(j)
        END IF
    END DO

END SUBROUTINE ST_LCUC

File: data/readable_format/cpp/file772.cpp
#include <iostream>
#include <string>

// Define the ST_LCUC function directly in this file
void ST_LCUC(const std::string& input, std::string& output, int& iret) {
    // Initialize the output string and status code
    output = input; // Copy input to output
    iret = 0; // Initialize return code to 0

    // Get the size of the output string
    size_t isize = output.length();

    // Iterate through each character of the string
    for (size_t i = 0; i < isize; ++i) {
        // Check if the character is lowercase
        if (output[i] >= 'a' && output[i] <= 'z') {
            // Convert to uppercase by subtracting 32 from ASCII value
            output[i] = output[i] - 32;
        }
    }
    // No explicit return needed in void function
}

// Example usage and a simple test case
int main() {
    std::string original = "Hello, World!";
    std::string converted;
    int status;
    
    // Call the ST_LCUC function with the original string
    ST_LCUC(original, converted, status);

    // Output the results
    std::cout << "Original: " << original << std::endl;
    std::cout << "Converted: " << converted << std::endl;

    // Simple check to verify if the conversion is correct
    if (converted == "HELLO, WORLD!" && status == 0) {
        std::cout << "Test passed." << std::endl;
    } else {
        std::cout << "Test failed." << std::endl;
    }

    return 0;
}